### ISSUE TITLE
Hardened builds on Fedora 23+ fail

### ISSUE THREAD

**Issue by belegdol** [Yabause/yabause#270]
Fedora 23 and above require all packages to use position-independent code [1]. As a result, yabause does not build anymore:
> /usr/bin/ld: CMakeFiles/yabause.dir/sh2_dynarec/linkage_x64.s.o: relocation R_X86_64_32S against `master_ip' can not be used when making a shared object; recompile with -fPIC

or, in static libs mode (-DBUILD_SHARED_LIBS:BOOL=OFF)
> /usr/bin/ld: ../libyabause.a(linkage_x64.s.o): relocation R_X86_64_32S against `master_ip' can not be used when making a shared object; recompile with -fPIC

The problem is that linkage_x64.s.o is assembled, not compiled, which means that it cannot be recompiled with -fPIC. According to a few stack overflow posts, assembly code needs to be written to be position-independent.

**Comment by Guillaumito** [comment]
Yeah, that's a problem with the dynarec... and there's no one maintaining this part of the code, so you'll have to disable it using the SH2_DYNAREC CMake variable. Something like:

cmake -DSH2_DYNAREC=OFF ...

**Comment by belegdol** [comment]
Thank you for the prompt answer! What would the performance hit be without the dynarec? I managed to work the issue around by disabling the hardening but if this code is not maintained it can clearly take time until it's fixed

**Comment by d356** [comment]
On my system running the bios the difference is about 105 fps -> 89 fps.
